stellargraph/core/tf_utils.py

```diff
@@ -0,0 +1,63 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+from ..core.experimental import experimental
+
+
+@experimental(reason="lack of unit tests")
+def partial_powers(one_hot_encoded_row, Aadj_T, num_powers=5):
+    """
+    This function computes the first num_powers powers of the adjacency matrix
+    for the row specified in one_hot_encoded_row
+
+    .. warning::
+
+        This function is experimental: it is insufficiently tested.
+
+    Args:
+        one_hot_encoded_row: one-hot-encoded row
+        Aadj_T: the transpose of the adjacency matrix
+        num_powers (int): the adjacency number of powers to compute
+
+    returns:
+        A matrix of the shape (num_powers, Aadj_T.shape[1]) of
+        the specified row of the first num_powers of the adjacency matrix.
+    """
+
+    # make sure the transpose of the adjacency is used
+    # tensorflow requires that the sparse matrix is the first operand
+
+    batch_adj = select_row_from_sparse_tensor(one_hot_encoded_row, Aadj_T)
+
+    partial_powers_list = [batch_adj]
+
+    for i in range(1, num_powers):
```

Comment:
Oh nice, that's neater! I'll change the loop body but it might still be useful to have a `select_row_from_sparse_tensor` function for future use.

And yeah, we don't want the zeroth power/identity matrix